### ISSUE TITLE
Add profile system for multiple user boxes

### ISSUE THREAD

**Issue by jeapostrophe** [libretro/RetroArch#4749]
The main purpose would be different recently played, some settings, retro achievements, and (most importantly) save files. Many games only allow one save per file, so would be convenient to make it easy to support this in RA.

**Comment by jeapostrophe** [comment]
pr's welcome

**Comment by Allariah** [comment]
Seconding this, was about to post a second issue but thankfully GitHub's smart issue search found it when I couldn't.

Here's my thought process with it:

----------------------------------------------------------------------------------------------
Multiple Users/Profiles like that of PS4/Plex ? (Or Windows/Mac/Linux but, less file duplication/bloat)

TL;DR OF BELOW: Same ROM library across all users, but multiple 'Save Profiles' to enable 2 people to play the same game on a given installation, without needing the game to support multiple save slots.


-------------------------------------------------------------------------------
My initial interpretation of the 'flow' would be:
On First Launch: Lakka asks if this installation is for a Single User, or Multi-Users, if it's a single user, the installation is _basically_ the exact same as it currently is. (Except ask the user for a name, or offer a default option, and the option to add/convert to a multi-user setup is now present in settings)
If selecting multi-users, then a dialog pops up asking you to enter a username, (and select an icon?) Rinse and repeat for as many users as you want.

The only issue I can see with this is confusing some people between Users and Players, so a small little information box may be helpful to explain the difference is to do with multiple people using the console independently to play the same game, not at the same time for multiplayer.

As for converting to a multi setup, as you now ask the user for a name on the first launch sequence, you can just add other named users and present the multi-user screen on boot instead of the default XMB.


Taking this a step further then... I'd love that you could have an 'Admin' user, who can update the system, edit cores, change SMB/SSH, turn on/off system settings etc, and all the 'Standard' users can see are the Games and Basic System Options for Sound etc.


------------------------------------------------------------
I'm running it directly as the Linux image on a small HTPC, and it's the only thing I can find that I 'dislike' about the setup. As I can't play the same game my brother wants to unless the game itself supports multiple save slots.

**Comment by natinusala** [comment]
Could you make an exhaustive list of what you would like separated exactly?

**Comment by jeapostrophe** [comment]
The most important things for me are

- Recently Played playlist
- Save States
- SRAMs

I can imagine wanting the ability to have user-specific preferences, like if I like a certain shader or if like a certain system remapping; but I don't personally want that. The other three things are the most important thing.

**Comment by natinusala** [comment]
I think we can do this cleanly by binding users to a specific config file, associated with a specific folder in `.config/retroarch` (or equivalent on your platform of choice). Each config file will have that users's settings as well as the path to their playlists / saves / games etc...

**Comment by Allariah** [comment]
Agreed, I'd rather the core functionality in-place for multi-users with the above listed by @jeapostrophe, with a small addition if possible (non-essential however).

- Individual Users
-- Recently Played
-- Save States 
-- SRAMs

- Guest/Temporary User
-- This never saves 'data', but allows/keeps applied cheats for things like fighting/party games to unlock characters/stages, then purges the playlists/save/savestates/srams on logout.

-----------------------------------------------------------------------------------------------

Beyond that in terms of a wishlist (I don't count the wishlist towards fulfillment of the bounty);
- User Classes
-- Admin (Full access to all settings)
-- Standard (Limited access to settings, based on config file?)
-- Guest (Same as standard but with the functionality listed above)

**Comment by natinusala** [comment]
If each user has their own settings then "user class" does not really matter does it

**Comment by Allariah** [comment]
> If each user has their own settings then "user class" does not really matter does it

Oh true, if they're completely separate entities that doesn't matter at all.

**Comment by askotx** [comment]
Could retroachievements be added to user profiles too?

**Comment by tralph3** [comment]
I made a program that accomplishes this if anyone is interested. It's not the same as adding it into RetroArch itself, but it gives the same functionality on a less convenient way: github.com/tralph3/Retroarch-User-Changer/releases

**Comment by Allariah** [comment]
Hi, Could the bounty label be added to this issue?

**Comment by swever826** [comment]
Will this feature be implemented someday? It would be a very useful feature in my opinion

**Comment by Dankakes** [comment]
Been trying to see if this is a default feature for like 2 days, seeing this issue is inspiring me to try to fix it myself and collect the bounty 3 years later haha

**Comment by jedenastka** [comment]
https://github.com/bountysource/core/issues/1586
